pennylane_qiskit/qiskit_device2.py

```diff
@@ -581,36 +556,49 @@ def _execute_runtime_service(self, circuits, session):
     def _execute_sampler(self, circuit, session):
         """Execution for the Sampler primitive"""
 
-        qcirc = circuit_to_qiskit(circuit, self.num_wires, diagonalize=True, measure=True)
-        if circuit.shots:
-            self.options.execution.shots = circuit.shots.total_shots
-        sampler = Sampler(session=session, options=self.options)
+        qcirc = [circuit_to_qiskit(circuit, self.num_wires, diagonalize=True, measure=True)]
+        sampler = Sampler(session=session)
+        compiled_circuits = self.compile_circuits(qcirc)
```

Comment:
Transpilation has been added in this PR. It was a separate PR before but since it is only a few lines (8 lines) and the tests will fail without transpilation I decided to do it in this PR.